core/distributed/matrix.cpp

```diff
@@ -42,6 +42,7 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 
 namespace gko {
+namespace experimental {
```

Comment:
I'm not sure about the inline namespace. Wouldn't that defeat the purpose of the namespace, because a user can just write use `gko::distributed::Vector` instead of `gko::experimental::distributed::Vector`? In the former case, it might not be entirely clear to the user that `Vector` for example might still be changed at any point.
Regarding the order, I think @upsj also used this order for the direct solver stuff, and at least for me, the grouping wrt. experimental is stronger that wrt. distributed.

Reply:
inline namespace is an option only when we think other information is enough for indicating it is experimental.
If not, we definitely keep it.
I am think of the order because something like Partition in distributed is not experimental.

Reply:
I guess with the move of Partition into experimental this should be fine.